FirebasePerformance/Sources/AppActivity/FPRSessionManager.m

```diff
@@ -57,41 +55,25 @@ - (FPRSessionManager *)initWithNotificationCenter:(NSNotificationCenter *)notifi
   self = [super init];
   if (self) {
     _sessionNotificationCenter = notificationCenter;
-    _trackingApplicationStateChanges = NO;
-    [self updateSessionId:nil];
+    _sessionDetails = [[FPRSessionDetails alloc] initWithSessionId:@"placeholder"
```

Comment:
Would there be a functionality change here? In the past, as a part of app startup, we used to generate a sessionID and start collecting gauges.

In this new world
   * Would there be delay in starting to collect gauges?
   * Secondly, starting with placeholder and then moving to an actual sessionID could be tricky since we might end up sending events with the placeholder sessionID. Anything we can do to make sure the gauges are not incorrectly referenced. For eg: creating a cache of events until we get a sessionID, skip data collection until we have a valid sessionID or could be any other alternatives.

Reply:
There would be no functionality change because we call `[sessions registerWithSubscriber:[self sharedInstance]];` in the `FIRComponentCreationBlock`. So as soon as Firebase creates Fireperf, `FirebaseSessions.register(subscriber)` will immediately call `onSessionChanged`, giving Fireperf a session ID. This is before any trace is yet created. 
![image](https://user-images.githubusercontent.com/12737815/213528078-a8c21cd6-3c96-451b-a230-7ab48e46da9b.png)

In my tests, `onSessionChanged` is always the first one to initialize the singleton `FPRSessionManager`, then immediately call `updateSessionId` as shown above, so actually the gauges are now collected earlier, since we did not have the creation block immediately initialize `FPRSessionManager` before.

Reply:
Let's make sure to comment these details above to give the context for any future developers.

Aside, instead of the placeholder, can we start with an empty string sessionID (instead of the text placeholder)?

Reply:
Done